tools/integration/test/integration/e2e-test-service/originsTest.js

```diff
@@ -0,0 +1,113 @@
+const assert = require('assert')
+const { callFetch } = require('../../../lib/fetch')
+const { devApiBaseUrl, prodApiBaseUrl, getComponents, definition } = require('../testConfig')
+
+const ORIGIN_EXCLUSION_LIST = ['go/golang', 'debsrc/debian', 'maven/mavengoogle']
+const ORIGIN_REVISIONS_EXCLUSION_LIST = ['debsrc/debian']
+
+const MAVEN_COMPONENT_GROUP_ID = 'maven/mavencentral/org.apache.httpcomponents'
+const MAVEN_COMPONENT_PARTIAL_GROUP_ID = 'maven/mavencentral/org.apache.httpcompon'
+const GRADLE_COMPONENT_ENDPOINT = 'gradleplugin/io.github.lognet.grpc-spring-boot'
+
+;(async function validateOriginsApi() {
+  const components = await getComponents()
+
+  describe('Validate Origins API Between Dev and Prod', function () {
+    this.timeout(definition.timeout)
+
+    afterEach(() => new Promise(resolve => setTimeout(resolve, definition.timeout / 2)))
+
+    components.filter(isOriginAllowed).forEach(component => {
+      it(`Validates Origins API response for ${component}`, () => compareOrigins(component))
+    })
+
+    components.filter(isOriginWithRevisionsAllowed).forEach(component => {
+      it(`Validates Origins API response with revisions for ${component}`, () => compareOriginsWithRevisions(component))
+    })
+
+    it('Validates Origins API response for a Maven component with only a group ID', () =>
+      compareOrigins(MAVEN_COMPONENT_GROUP_ID))
+
+    it('Validates Origins API response for a Maven component with a partial group ID for suggestion checks', () =>
+      compareOrigins(MAVEN_COMPONENT_PARTIAL_GROUP_ID))
+
+    it('Validates Origins API response for a Gradle plugin component', () =>
```

Comment:
why do we have to separate gradle components?

Reply:
I have used separate test cases for gradle since the endpoint for it is origins/gradleplugin/:pluginID where pluginID is not consistent with the name of group id or artefact id like 

Here is the case where pluginId is combination of group id and artefact id
https://plugins.gradle.org/plugin/io.outfoxx.sunday-generator
https://mvnrepository.com/artifact/io.outfoxx.sunday/generator

Here is the case where pluginId is only group id -
https://plugins.gradle.org/plugin/io.github.lognet.grpc-spring-boot
https://mvnrepository.com/artifact/io.github.lognet.grpc-spring-boot/io.github.lognet.grpc-spring-boot.gradle.plugin

Please let me know if I could handle this better. Most of the cases corresponds pluginId to group id. In that case we would also need to add another buildurl function for gradleplugin.